### PR TITLE
Docker image for SimpleVote

### PR DESCRIPTION
**Pull request by bosonogi**

### CODE REVIEW

**Comment by bosonogi** [comment]
Noticed some issues with this; will work on it more.

**Comment by dessalines** [comment]
This look great! Thanks. 

It still seems kinda weird to me that the docker builds from scratch when the jars are already built out [here](https://github.com/dessalines/simple-vote/releases) tho.

So using the `java -jar simplevote.jar -liquibase` (and your environment variables change, or the properties file) will already have everything you need, since the front end resources are already bundled in there too. I spose docker does automate installing java, and hooking it up to a postgres DB, so that's nice. 

Basically the whole point of docker is portability, but the jar is already *nearly* as portable as you can get.

**Comment by dessalines** [comment]
I don't use docker, but just keep pushing changes to this branch and I'll merge them as you fix em.

**Comment by bosonogi** [comment]
@dessalines 
> It still seems kinda weird to me that the docker builds from scratch

That is part of the point. While the jar gives you a portable application to run, the Dockerfile gives you portability for building it -- only a docker daemon is needed (no manual jdk, maven, node and angular setup needed).

Unfortunately I ran out of time for tinkering with this at the moment; hopefully I will be able to come back to it at a later date. Here are the issues that I've encountered:
- When starting form a clean git clone, `docker-compose up` fails with a `java.io.FileNotFoundException: /simplevote.properties` exception.
- Frontend hardcodes absolute endpoints in its environment setup; there should be a way to avoid this.
- Related to the above and some angular specifics, there are problems serving the app from behind an nginx reverse proxy with a path prefix (eg. `https://<host>/simplevote`).
- It would be nice to be able to turn off chat in the frontend configuration. There a situations where it is unnecessary, and turning it off would simplify a reverse proxy configuration as it wouldn't need to proxy web sockets.

**Comment by dessalines** [comment]
This should be fixed now.
